services/core/shared/indexdb/mysql.js

```diff
@@ -0,0 +1,183 @@
+	return val;
+};
+
+const getDbInstance = async (tableName, tableConfig, connEndpoint = config.endpoints.mysql) => {
+	const userName = connEndpoint.split('//')[1].split('@')[0].split(':')[0];
+	const [hostPort, dbName] = connEndpoint.split('@')[1].split('/');
+	const connPoolKey = `${userName}@${hostPort}/${dbName}`;
+	const connPoolKeyTable = `${connPoolKey}/${tableName}`;
+
+	if (!connectionPool[connPoolKey]) {
+		connectionPool[connPoolKey] = await createDb(connEndpoint);
+	}
+
+	const knex = connectionPool[connPoolKey];
+
+	if (!tablePool[connPoolKeyTable]) {
+		await loadSchema(knex, tableName, tableConfig);
+		tablePool[connPoolKeyTable] = true;
+	}
+
+	const { primaryKey, schema } = tableConfig;
+
+	const upsert = async (inputRows) => {
+		let rawRows = inputRows;
+		if (!Array.isArray(rawRows)) rawRows = [inputRows];
+
+		const rows = [];
+		rawRows.forEach(item => {
+			const row = {};
+			Object.keys(schema).forEach(o => {
+				row[o] = getValue(cast(item[o], schema[o].type));
+			});
+			rows.push(row);
+		});
+
+		try {
+			const chunkSize = 1000;
+			const ids = await knex.batchInsert(tableName, rows, chunkSize);
+			logger.debug(`${rows.length} row(s) inserted/updated in '${tableName}' table`);
+			return ids;
+		} catch (error) {
+			const errCode = error.code;
+			const errMessage = error.message.split(`${errCode}: `)[1] || error.message;
+			const errCause = errMessage.split(': ')[0];
+			logger.debug('Encountered error with batch insert:', errCause,
+				'\nRe-attempting to update/merge the conflicted transactions one at a time: ');
+
+			return knex.transaction(async trx => {
+				// TODO: Consider replacing promise all to Bluebird
+				const inserts = await Promise.all(rows.map(row => trx(tableName).insert(row).onConflict('id').merge()
+					.transacting(trx)));
+				logger.debug(`${rows.length} row(s) inserted/updated in '${tableName}' table`);
+				return inserts;
+			});
+		}
+	};
+
+	const queryBuilder = (params, columns) => {
+		const limit = params.limit || 1;
+		const offset = params.offset || 0;
+
+		delete params.limit;
+		delete params.offset;
+
+		const query = knex.select(columns).table(tableName);
+
+		if (params.propBetween) {
+			const { propBetween } = params;
+			delete params.propBetween;
+			query.whereBetween(propBetween.property, [propBetween.from, propBetween.to]);
+		}
+
+		if (params.orWhere) {
+			const { orWhere } = params;
+			delete params.orWhere;
+			query.orWhere(orWhere);
+		}
+
+		if (params.sort) {
+			const [sortProp, sortOrder] = params.sort.split(':');
+			delete params.sort;
+			query.orderBy(sortProp, sortOrder);
+		}
+
+		return query.andWhere(params)
+			.limit(limit)
+			.offset(offset);
+	};
+
+	const find = (params, columns) => new Promise((resolve, reject) => {
+		queryBuilder(params, columns).then(response => {
+			resolve(response);
+		}).catch(err => {
+			logger.error(err.message);
+			reject(err);
+		});
+	});
+
+	const deleteIds = async ids => knex(tableName)
```

Comment:
A more generic `delete` method could be put in place that builds atop the `find` method (or `queryBuilder`) and the resultSet chained by `del()`.

```suggestion
	const deleteIds = async ids => knex(tableName)
```

Reply:
The problem with delete is that `new` and `delete` are reserved keywords in JS and they confuse linter. I still doubt whether we really should use that. I kind of works, but still we need to avoid doing it I think.

https://www.w3schools.com/js/js_reserved.asp

Reply:
Perhaps `findAndDelete`could be our way around?

Reply:
I was thinking about it, although it is not needed yet.
We can implement on the first occurence.

Reply:
Yes, sure. It was a recommendation for the future.